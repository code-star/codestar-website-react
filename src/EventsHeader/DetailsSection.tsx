import React, { SFC } from 'react';
import { Typography, Button } from '@material-ui/core';
import { translate, TranslationFunction } from 'react-i18next';
import Container from '../Container/Container';
import Section from '../Section/Section';
import compose from 'recompose/compose';

// TODO improve types by replacing "any"
interface IDetailsSectionPropsInner {
	t: TranslationFunction;
	mEvent: any;
	formattedDate: string;
	descriptionElem: any;
}

interface IDetailsSectionPropsOuter {
	mEvent: any;
	formattedDate: string;
	descriptionElem: any;
}

export const DetailsSection: SFC<IDetailsSectionPropsInner> = ({
	t,
	mEvent,
	formattedDate,
	descriptionElem,
}) => {
	return (
		<Section scrollname="event-details" className="bg-white">
			<Container center>
				<div className="row">
					<div className="col-12">
						<Typography align="center" variant="title">
							{mEvent.name}
						</Typography>
						<Typography gutterBottom align="center" variant="subheading">
							{formattedDate}
						</Typography>
					</div>
				</div>
				<div className="row">
					<div className="col-12 col-md-8">{descriptionElem}</div>
					<div className="col-12 col-md-4">
						<img
							src={mEvent.coverUrl}
							alt={`Artistic background with text "${mEvent.name}"`}
							style={{ width: '100%' }}
						/>
						<Button
							color="primary"
							variant="raised"
							href={mEvent.link}
							className="mt-1"
						>
							{t('SIGN_UP')}
						</Button>
					</div>
				</div>
			</Container>
		</Section>
	);
};

export default compose<IDetailsSectionPropsInner, IDetailsSectionPropsOuter>(
	translate(['events'], { wait: true })
)(DetailsSection);
