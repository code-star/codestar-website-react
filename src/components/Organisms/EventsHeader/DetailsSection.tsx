import React, { SFC, ReactNode } from 'react';
import { Typography, Button } from '@material-ui/core';
import { translate, TranslationFunction } from 'react-i18next';
import Container from '../../../Container/Container';
import Section from '../../Molecules/Section/Section';
import compose from 'recompose/compose';
import TweetList from '../../Molecules/List/TweetList';
import {
  IMeetupEvent,
  ITweet,
} from '../../../containers/EventsContainer/EventsContainer.interfaces';

interface IDetailsSectionPropsInner {
  t: TranslationFunction;
}

interface IDetailsSectionPropsOuter {
  mEvent: IMeetupEvent;
  formattedDate: string;
  descriptionElem: ReactNode;
  tweets: ITweet[];
}

export const DetailsSection: SFC<
  IDetailsSectionPropsInner & IDetailsSectionPropsOuter
> = ({ t, mEvent, formattedDate, descriptionElem, tweets }) => {
  return (
    <Section scrollname="event-details" className="bg-white">
      <Container center>
        <div className="row">
          <div className="col-12 col-md-8">
            <Typography align="center" variant="h6">
              {mEvent.name}
            </Typography>
            <Typography gutterBottom align="center" variant="subtitle1">
              {formattedDate}
            </Typography>
            {descriptionElem}
          </div>
          <div className="col-12 col-md-4">
            <TweetList
              tweets={tweets}
              eventDate={formattedDate}
              eventImage={mEvent.coverUrl}
              eventName={mEvent.name}
            >
              <Button color="primary" href={mEvent.link}>
                {t('SIGN_UP')}
              </Button>
            </TweetList>
          </div>
        </div>
      </Container>
    </Section>
  );
};

export default compose<
  IDetailsSectionPropsInner & IDetailsSectionPropsOuter,
  IDetailsSectionPropsOuter
>(translate(['events'], { wait: true }))(DetailsSection);
