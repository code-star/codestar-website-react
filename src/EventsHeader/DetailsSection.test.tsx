import React from 'react';
import { DetailsSection } from './DetailsSection';
import renderer from 'react-test-renderer';

// Mock for "t", the translate function from react-i18next
const t = () => '';

it('renders the details for the Events header', () => {
	const meetupEvent = {
		name: 'Data Oriented Design with Maxim Zaks',
		link: 'https://www.codestar.nl',
	};
	const formattedDate = '26 september 2018';
	const comp = renderer
		.create(
			<DetailsSection
				t={t}
				mEvent={meetupEvent}
				formattedDate={formattedDate}
				descriptionElem={'descriptionElem'}
			>
				<button />
			</DetailsSection>
		)
		.toJSON();
	expect(comp).toMatchSnapshot();
});
