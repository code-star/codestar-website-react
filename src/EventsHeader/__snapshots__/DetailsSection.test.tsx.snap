// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the details for the Events header 1`] = `
<section
  className="
			bg-white
			WithTheme-WithWidth-Section---section-1"
  name="event-details"
>
  <div
    className="WithTheme-WithWidth-Section---element-2"
    name="event-details"
  />
  <div
    className="
			container
			
			
			
			
			
			WithTheme-WithWidth-Container---containerCenter-5
		"
  >
    <div
      className="row"
    >
      <div
        className="col-12"
      >
        <h2
          className="MuiTypography-root-8 MuiTypography-title-14 MuiTypography-alignCenter-21"
        >
          Data Oriented Design with Maxim Zaks
        </h2>
        <h3
          className="MuiTypography-root-8 MuiTypography-subheading-15 MuiTypography-gutterBottom-25 MuiTypography-alignCenter-21"
        >
          26 september 2018
        </h3>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-12 col-md-8"
      >
        descriptionElem
      </div>
      <div
        className="col-12 col-md-4"
      >
        <img
          alt="Artistic background with text \\"Data Oriented Design with Maxim Zaks\\""
          src={undefined}
          style={
            Object {
              "width": "100%",
            }
          }
        />
        <a
          className="MuiButtonBase-root-59 MuiButton-root-33 MuiButton-contained-44 MuiButton-containedPrimary-45 MuiButton-raised-47 MuiButton-raisedPrimary-48 mt-1"
          href="https://www.codestar.nl"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex="0"
        >
          <span
            className="MuiButton-label-34"
          >
            
          </span>
          <span
            className="MuiTouchRipple-root-62"
          />
        </a>
      </div>
    </div>
  </div>
</section>
`;
